pkg/apis/pipeline/v1alpha1/cloud_event_resource.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2019 The Tekton Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// CloudEventResource is an event sink to which events are delivered when a TaskRun has finished
+type CloudEventResource struct {
+	Name      string               `json:"name"`
+	Type      PipelineResourceType `json:"type"`
+	TargetURI string               `json:"targetURI"`
+	Owner     TaskRun              `json:"taskRun"`
```

Comment:
I'm interested in more detail about what functionality requires this `Owner` field - I think at the moment other PipelineResources don't have any fields that are explicitly tied to TaskRuns - and I could see a world where we want to use PipelineResources outside of TaskRuns (e.g. in triggers, conditionals)

I think I'm also a bit confused cuz the way this is declared it seems like the user would set this but I'm thinking that isn't the case (probably b/c I asked you to split this out and am now missing the context haha) - maybe this is an implementation detail? In which case I wonder if it can be unexported? Or maybe just a comment to explain it is enough at this point :)

Reply:
I think `Owner` dates back to the initial implementation where I was sending the cloud event from a container in the TaskRun pod as opposed to the controller as it is today. It is not actually used in the current implementation, so I will clean it up. Nice catch, thank you!